src/components/enterprise-user-subsidy/tests/UserSubsidy.test.jsx

```diff
@@ -0,0 +1,218 @@
+    expect(fetchSubscriptionLicensesForUser).toHaveBeenCalledWith(TEST_SUBSCRIPTION_UUID);
+
+    await waitFor(() => {
+      expect(screen.getByTestId('did-i-render')).toBeInTheDocument();
+    });
+
+    // assert component is no longer loading
+    expect(screen.queryByText(LOADING_SCREEN_READER_TEXT)).not.toBeInTheDocument();
+  });
+
+  test('renders license activation alert if user has an assigned (pending) license on Dashboard page route', async () => {
+    const promise = Promise.resolve({
+      data: {
+        results: [{
+          uuid: TEST_LICENSE_UUID,
+          status: LICENSE_STATUS.ASSIGNED,
+        }],
+      },
+    });
+    fetchSubscriptionLicensesForUser.mockResolvedValueOnce(promise);
+
+    const Component = <UserSubsidyWithAppContext contextValue={contextValue} />;
+    renderWithRouter(Component, {
+      route: `/${TEST_ENTERPRISE_SLUG}`,
+    });
+
+    // assert status alert message renders
+    await waitFor(() => {
+      const activationMessage = 'activate your enterprise license';
+      expect(screen.queryByRole('alert')).toBeInTheDocument();
+      expect(screen.queryByText(activationMessage, { exact: false })).toBeInTheDocument();
+    });
+  });
+
+  test('renders license deactivation alert if user has a deactivated (revoked) license on Dashboard page route', async () => {
+    const promise = Promise.resolve({
+      data: {
+        results: [{
+          uuid: TEST_LICENSE_UUID,
+          status: LICENSE_STATUS.DEACTIVATED,
+        }],
+      },
+    });
+    fetchSubscriptionLicensesForUser.mockResolvedValueOnce(promise);
+
+    const Component = <UserSubsidyWithAppContext contextValue={contextValue} />;
+    renderWithRouter(Component, {
+      route: `/${TEST_ENTERPRISE_SLUG}`,
+    });
+
+    // assert status alert message renders
+    await waitFor(() => {
+      const deactivationMessage = 'enterprise license is no longer active';
+      expect(screen.queryByRole('alert')).toBeInTheDocument();
+      expect(screen.queryByText(deactivationMessage, { exact: false })).toBeInTheDocument();
+    });
+  });
+
+  test('renders unassigned license alert if user does not have an associated license on Dashboard page route', async () => {
+    const promise = Promise.resolve({
+      data: {
+        results: [],
+      },
+    });
+    fetchSubscriptionLicensesForUser.mockResolvedValueOnce(promise);
+
+    const Component = <UserSubsidyWithAppContext contextValue={contextValue} />;
+    renderWithRouter(Component, {
+      route: `/${TEST_ENTERPRISE_SLUG}`,
+    });
+
+    // assert status alert message renders
+    await waitFor(() => {
+      const deactivationMessage = 'do not have an enterprise license';
+      expect(screen.queryByRole('alert')).toBeInTheDocument();
+      expect(screen.queryByText(deactivationMessage, { exact: false })).toBeInTheDocument();
+    });
+  });
+
+  test('redirects to Dashboard page if user has an assigned (pending) license on non-Dashboard page route', async () => {
```

Comment:
This makes sense for our current pages, I'm just wondering if there will be any finickiness when we have the activation route where a user gets redirected after the request.
I wouldn't change anything, just making a mental note

Reply:
Yeah, that's a good callout. I imagine the implementation ticket for the activation route will have to decide whether to use/modify `UserSubsidy` or do a standalone implementation (i.e., its own API call through the `useSubscriptionLicenseForUser` hook).